src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_exit_application_notification.cc

```diff
@@ -112,6 +112,15 @@ void OnExitApplicationNotification::Run() {
       break;
     }
 #endif  // CLOUD_APP_WEBSOCKET_TRANSPORT_SUPPORT
+    case Common_ApplicationExitReason::RESOURCE_CONSTRAINT: {
+      const auto message =
+          MessageHelper::GetOnAppInterfaceUnregisteredNotificationToMobile(
+              app_id, AppInterfaceUnregisteredReason::RESOURCE_CONSTRAINT);
+      SendNotificationToMobile(message);
+      application_manager_.UnregisterApplication(app_id, Result::SUCCESS);
+      application_manager_.connection_handler().CloseRevokedConnection(app_id);
```

Comment:
@mvorobio What is the reason to call this function?

Reply:
@AByzhynar CloseRevokedConnection is called to close WebSocket connection associated to the web app HMI wants to exit.

Reply:
@mvorobio I understand what the function does. I am asking why did you decide to close connection? I can't see any requirements.  So for me it looks just doing something which is not required and requested by the customer.

Reply:
@AByzhynar Summary of https://adc.luxoft.com/jira/browse/FORDTCN-5000 implies this. Not sure if it is correct interpretation of "there is a way for the HMI to be able to unregister and **disconnect** an app in case of performance issues"

Reply:
@mvorobio Don't interpret. Interpretations can differ. Read it as is: `unregister and disconnect an app` but you disconnect the entire transport, don't you?

Reply:
@AByzhynar @mvorobio 
I believe, in this case we disconnect only session for this app, other sessions and listener will be continue working.